src/workflows/methods/modals/EditWorkflowModal.test.tsx

```diff
@@ -0,0 +1,327 @@
+import { EditMethodProvider } from 'src/libs/ajax/methods/providers/EditMethodProvider';
+import { renderWithAppContexts as render } from 'src/testing/test-utils';
+import { EditWorkflowModal } from 'src/workflows/methods/modals/EditWorkflowModal';
+
+type WDLEditorExports = typeof import('src/workflows/methods/WDLEditor');
+jest.mock('src/workflows/methods/WDLEditor', (): WDLEditorExports => {
+  const mockWDLEditorModule = jest.requireActual('src/workflows/methods/WDLEditor.mock');
+  return {
+    WDLEditor: mockWDLEditorModule.MockWDLEditor,
+  };
+});
+
+const mockCreateSnapshotResponse: MethodResponse = {
+  name: 'my-workflow',
+  createDate: '2024-11-22T11:22:44Z',
+  documentation: 'documentation',
+  synopsis: 'synopsis',
+  entityType: 'Workflow',
+  snapshotComment: 'snapshot comment',
+  snapshotId: 4,
+  namespace: 'my-namespace',
+  payload: 'workflow doStuff {}',
+  url: 'https://agora.dsde-dev.broadinstitute.org/api/v1/methods/my-namespace/my-workflow/4',
+};
+
+const editMethodProviderError: EditMethodProvider = {
+  createNewSnapshot: jest.fn(() => {
+    throw new Error('Error thrown for testing purposes!');
+  }),
+};
+
+const editMethodProviderErrorResponse: EditMethodProvider = {
+  createNewSnapshot: jest.fn(() => {
+    throw new Response('Error response thrown for testing purposes!');
+  }),
+};
+
+const editMethodProviderSuccess: EditMethodProvider = {
+  createNewSnapshot: jest.fn().mockResolvedValue(mockCreateSnapshotResponse),
+};
+
+describe('EditWorkflowModal', () => {
+  it('renders elements with given values', async () => {
+    // Act
+    await act(async () => {
+      render(
+        <EditWorkflowModal
+          title='Edit'
+          namespace='my-namespace'
+          name='my-workflow'
+          snapshotId={3}
+          defaultWdl='workflow doStuff {}'
+          defaultDocumentation='documentation'
+          defaultSynopsis=''
+          editMethodProvider={editMethodProviderSuccess}
+          onSuccess={jest.fn()}
+          onDismiss={jest.fn()}
+        />
+      );
+    });
+
+    // Assert
+    expect(screen.getByRole('textbox', { name: 'Namespace' })).toHaveAttribute('placeholder', 'my-namespace');
+    expect(screen.getByRole('textbox', { name: 'Namespace' })).toHaveAttribute('disabled');
+    expect(screen.getByRole('textbox', { name: 'Name' })).toHaveAttribute('placeholder', 'my-workflow');
+    expect(screen.getByRole('textbox', { name: 'Name' })).toHaveAttribute('disabled');
+    expect(screen.getByTestId('wdl editor')).toHaveDisplayValue('workflow doStuff {}');
+    expect(screen.getByRole('textbox', { name: 'Documentation' })).toHaveDisplayValue('documentation');
+    expect(screen.getByRole('textbox', { name: 'Synopsis (80 characters max)' })).toHaveDisplayValue('');
+    expect(screen.getByRole('textbox', { name: 'Snapshot comment' })).toHaveDisplayValue('');
+  });
+
+  it('successfully creates a new snapshot with inputted information', async () => {
```

Comment:
It wouldn't hurt to add a test that makes sure that by default `redactPreviousSnapshot` is set to `false` in the provider function call, given the consequences if it were always `true` regardless of whether the checkbox was checked.

Reply:
Good point. I have added that test.